src/debug.c

```diff
@@ -2317,24 +2348,6 @@ void serverLogHexDump(int level, char *descr, void *value, size_t len) {
 /* =========================== Software Watchdog ============================ */
 #include <sys/time.h>
 
-void sigalrmSignalHandler(int sig, siginfo_t *info, void *secret) {
-#ifdef HAVE_BACKTRACE
-    ucontext_t *uc = (ucontext_t*) secret;
-#else
-    (void)secret;
-#endif
-    UNUSED(info);
-    UNUSED(sig);
-
-    serverLogFromHandler(LL_WARNING,"\n--- WATCHDOG TIMER EXPIRED ---");
-#ifdef HAVE_BACKTRACE
-    logStackTrace(getAndSetMcontextEip(uc, NULL), 1);
-#else
-    serverLogFromHandler(LL_WARNING,"Sorry: no support for backtrace().");
-#endif
-    serverLogFromHandler(LL_WARNING,"--------\n");
-}
```

Comment:
did you move this so that it's not under the `======== Software Watchdog =======` title?
i think i'd rather avoid moving big blocks, and add forward declarations if needed.
i think it's arguably still ok to b considered part of the "Software Watchdog" code, and the fact it also serves another mechanism is ok.

Reply:
yes and also I wanted to group all the signal setups together

Reply:
i think now it's better.. also easier to see what's actually changed (the info->si_pid part)